src/ucp/rndv/proto_rndv.inl

```diff
@@ -45,13 +45,16 @@ ucp_proto_rndv_rts_request_init(ucp_request_t *req)
         return status;
     }
 
-    status = ucp_datatype_iter_mem_reg(ep->worker->context,
-                                       &req->send.state.dt_iter, rpriv->md_map,
-                                       UCT_MD_MEM_ACCESS_RMA |
-                                       UCT_MD_MEM_FLAG_HIDE_ERRORS,
-                                       UCS_BIT(UCP_DATATYPE_CONTIG));
-    if (status != UCS_OK) {
-        return status;
+    if (!UCP_DT_IS_GENERIC(req->send.state.dt_iter.dt_class)) {
```

Comment:
since ucp_datatype_iter_mem_reg already checks the dt_class, maybe we can pass some parameter to it whether to print error or not? or just not fail if UCT_MD_MEM_FLAG_HIDE_ERRORS is set?